projects/dataset_recorder/data_recorder.cc

```diff
@@ -0,0 +1,246 @@
+
+    const char *c  = folderName.c_str();
+    mkdir(c, 0777);
+    std::cout << "directory " << c << " created" << std::endl;
+
+    auto cam = getPanoramicCamera();
+    const auto cameraRes = cam->getOutputSize();
+    std::cout << "camera initialised" << std::endl;
+
+    // unwrapper for camera
+   // auto unwrapper = cam->createUnwrapper(cv::Size(WIDTH,HEIGHT));
+
+
+    cv::Mat imgOrig;
+    //--------------------------------->>>>>>>>>>> START RECORDING <<<<<<<<<<<<----------------------------
+
+
+    // start camera thread
+    cv::Mat tmpImage, unwrappedImage, resized;
+    cv::Mat outputImage(cv::Size(WIDTH,HEIGHT), CV_8UC3);
+    std::thread readCameraThread(&readCameraThreadFunc, cam.get(), std::ref(tmpImage), std::ref(shouldQuit), std::ref(mex));
+    std::this_thread::sleep_for(std::chrono::milliseconds(3000));
+
+    // create and open file to write gps coordinates
+    std::ofstream coordinates;
+
+    //write headers  for csv
+    std::string csvpath = folderName + "/coordinates.csv";
+    coordinates.open (csvpath, std::ofstream::trunc);
+    coordinates  << "gps quality" << "," << "lat" << "," << "lon" << "," << "altitude"  << "," << "roll" << "," << "pitch" << "," << "yaw"  << ",";
+    coordinates << "image name" << "," << "timestamp" <<"\n";
+    coordinates.close();
+
+
+    BoBRobotics::BackgroundExceptionCatcher catcher;
+    catcher.trapSignals();
+
+    int i = 0;
+    BoBRobotics::Stopwatch sw, sw_timestamp;
+    std::array<degree_t, 3> angles;
+    // for x timestep
+    sw_timestamp.start();
+    for (;;) {
+
+        try {
+            catcher.check();
+            sw.start();
+            // get imu data
+            angles = imu.getEulerAngles();
+            float yaw = angles[0].value();
+            float pitch = angles[1].value();
+            float roll = angles[2].value();
+
+            // get gps data
+            BoBRobotics::GPS::GPSData data = gps.getGPSData();
+            BoBRobotics::MapCoordinate::GPSCoordinate coord = data.coordinate;
+            BoBRobotics::GPS::TimeStamp time = data.time;
+            int gpsQual = (int) data.gpsQuality; // gps quality
+
+            // output results
+    	   // if (i%5 == 0) { // only output once ever sec
+            std::cout << std::setprecision(10) << "GPS quality: " << gpsQual << " latitude: " << coord.lat.value()<< " longitude: " <<  coord.lon.value()  << " num sats: " << data.numberOfSatellites << " time: " << static_cast<units::time::millisecond_t>(sw_timestamp.elapsed()).value()/1000 << std::endl;
+
+	        std::ostringstream fileString, folderString;
+            fileString << folderName << "/image" << i << ".jpg";
+	        folderString << folderName << "/coordinates.csv";
+            std::string fileName = fileString.str();
+
+            //  headers:   gpsQuality|latitude|longitude|altitude|roll|pitch|yaw|imagename|timestamp
+            coordinates.open (folderString.str(), std::ofstream::app); // open coordinates.csv file and append
```

Comment:
why are you re-opening the file here? can't you open it once outside of the loop?

Reply:
I think this is an important one - it's a very simple fix and constantly opening and closing files will prevent buffering which will really thrash the poor Jetson's shitty filesystem...